lib/server.js

```diff
@@ -32,6 +32,10 @@ var Router = util.safeRequire('react-router');
 var match = Router.match;
 var RoutingContext = Router.RoutingContext;
 
+//for creating a URL
+var createMemoryHistory = require('history').createMemoryHistory;
+var createLocation = createMemoryHistory().createLocation;
```

Comment:
`thing` is `req.url`. Since we are inside the renderer, keep in mind we don't have access to req or res